src/core/net/ip6_address.cpp

```diff
@@ -114,7 +114,7 @@ bool Address::IsRealmLocalAllMplForwarders(void) const
 bool Address::IsRoutingLocator(void) const
 {
     return (mFields.m16[4] == HostSwap16(0x0000) && mFields.m16[5] == HostSwap16(0x00ff) &&
-            mFields.m16[6] == HostSwap16(0xfe00));
+            mFields.m16[6] == HostSwap16(0xfe00) && (mFields.m8[14] & 0x02) == 0 && mFields.m8[14] < 0xfc);
```

Comment:
It would be better to use `mFields.m8[14] != Mle::kAloc16Mask`.

There is a similar function: [IsRoutingLocator()](https://github.com/openthread/openthread/blob/master/src/core/thread/mle.cpp#L3197), I think we can remove it since they are duplicate, as well as [IsAnycastLocator()](https://github.com/openthread/openthread/blob/master/src/core/thread/mle.cpp#L3202).